server/mscalendar/availability.go

```diff
@@ -157,7 +159,7 @@ func (m *mscalendar) setUserStatuses(users []*store.User, calendarViews []*remot
 	numberOfLogs, numberOfUserStatusChange, numberOfUserErrorInStatusChange := 0, 0, 0
 	toUpdate := []*store.User{}
 	for _, u := range users {
-		if u.Settings.UpdateStatus {
+		if u.Settings.UpdateStatusFromOptions != NotSetStatusOption || u.Settings.SetCustomStatus {
```

Comment:
@mickmister I think there is some confusion here.

- There were two settings before:
   1. If a user needs to change their status or not.
   2. If yes, it should be away or DND.

- In the new flow we have s single setting for this. The user gets three options in a single setting.
   1. Away
   2. DND
   3. Do not set
   
Because of the new flow the `UpdateStatus` field does not exist anymore